File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    const hayLen = haystack.length;
    const needleLen = needle.length;

    if (needleLen === 0) {
        return 0;
    }

    for (let i = 0; i <= hayLen - needleLen; i++) {
        if (haystack.slice(i, i + needleLen) === needle) {
            return i;
        }
    }

    return -1;
};

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == "":
            return 0

        hay_len = len(haystack)
        needle_len = len(needle)

        for i in range(hay_len - needle_len + 1):
            if haystack[i:i + needle_len] == needle:
                return i

        return -1
    